modules/oe_corporate_countries_address/tests/src/Kernel/AvailableCountriesSubscriberTest.php

```diff
@@ -0,0 +1,310 @@
+  ];
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The state service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * Name of the field.
+   *
+   * @var string
+   */
+  protected $fieldName = 'field_country';
+
+  /**
+   * Name of the field type.
+   *
+   * @var string
+   */
+  protected $fieldTypeName = 'address_country';
+
+  /**
+   * Name of the entity type.
+   *
+   * @var string
+   */
+  protected $entityTypeName = 'entity_test';
+
+  /**
+   * Name of the bundle to be tested.
+   *
+   * @var string
+   */
+  protected $bundleName = 'oe_countries';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installEntitySchema('configurable_language');
+    $this->installConfig(['language']);
+    // At least one extra language needs to be enabled for translations to work.
+    ConfigurableLanguage::createFromLangcode('it')->save();
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+    $this->state = $this->container->get('state');
+
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('entity_test');
+
+    $entity = EntityTest::create();
+
+    FieldStorageConfig::create([
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'type' => $this->fieldTypeName,
+    ])->save();
+
+    FieldConfig::create([
+      'label' => 'My field',
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'bundle' => $this->entityTypeName,
+    ])->save();
+
+    $entity_form_display = EntityFormDisplay::collectRenderDisplay($entity, 'default');
+
+    $entity_form_display->setComponent($this->fieldName, [
+      'region' => 'content',
+      'type' => 'address_country_default',
+    ]);
+    $entity_form_display->save();
+
+  }
+
+  /**
+   * Tests the deprecated countries are removed from the countries array.
+   * */
+  public function testGetAvailableCountriesArray(): void {
+
+    $values = [
+      $this->fieldName => [
+        'value' => 'FR',
+      ],
+    ];
+
+    $entity = EntityTest::create($values);
+    $entity->save();
+
+    $entity = $this->reloadEntity($entity);
+
+    /** @var \Drupal\field\Entity\FieldStorageConfig $field_storage */
+    $field_storage = $entity->getFieldDefinitions()[$this->fieldName]
+      ->getFieldStorageDefinition();
+
+    $field_storage->getPropertyDefinitions();
+
+    // Get available_countries from field.
+    // This doesn't have the deprecated countries.
+    $form = $this->buildEntityTestForm();
+    $availableCountries = $form["field_country"]["widget"][0]["value"]["#available_countries"];
+
+    // Get all the countries including the deprecated countries from sparql db.
+    $allCountries = $this->getAllCountries();
```

Comment:
The tests use a set list of countries, so we always know what the value of $availableCountries needs to be: ["XK", "AX", "BE", "TF", "IT"]. The test just needs to assert that the value of $availableCountries is equal to that set array. (See similar case here https://github.com/openeuropa/oe_corporate_countries/blob/2.x/modules/oe_corporate_countries_address/tests/src/Kernel/CountryRepositoryTest.php#L49)